File: engine/engine-model/src/main/scala/tools/packet/message/ItemMegaphone.scala
package tools.packet.message

import client.inventory.Item
import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ItemMegaphone(private var _message: String, private var _whisper: Boolean, private var _channel: Int,
                    private var _item: Item) extends PacketInput {
  def message: String = _message

  def whisper: Boolean = _whisper

  def channel: Int = _channel

  def item: Item = _item

  override def opcode(): SendOpcode = SendOpcode.SERVER_MESSAGE
}

File: engine/engine-model/src/main/java/tools/packet/message/ItemMegaphone.java
package tools.packet.message;

import client.inventory.Item;
import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ItemMegaphone(String message, Boolean whisper, Integer channel, Item item) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.SERVER_MESSAGE;
   }
}